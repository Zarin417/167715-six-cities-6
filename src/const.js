export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const CardName = {
  CITIES: `cities`,
  FAVORITES: `favorites`,
  NEAR_PLACES: `near-places`,
};

export const CardsListName = {
  NEAR_PLACES_LIST: `near-places__list`,
  CITIES_PLACES_LIST: `cities__places-list tabs__content`
};

export const ButtonName = {
  PROPERTY: `property`,
  PLACE_CARD: `place-card`,
  FAVORITE: `favorite`,
  NEAR_PLACE: `near-place`
};

export const SortType = {
  POPULAR: `popular`,
  PRICE_LOW_TO_HIGH: `price: low to high`,
  PRICE_HIGH_TO_LOW: `price: high to low`,
  TOP_RATED: `top rated first`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  CITY: `/city/:city`,
  OFFER: `/offer/:id`,
  ERROR: `/error`
};

export const APIRoute = {
  HOTELS: `/hotels`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};
