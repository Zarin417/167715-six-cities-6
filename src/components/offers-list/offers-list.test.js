import React from "react";
import {render} from "@testing-library/react";
import {createMemoryHistory} from "history";
import configureStore from "redux-mock-store";
import {Router} from "react-router";
import {CardsListName} from "../../const";
import {Provider} from "react-redux";
import OffersList from "./offers-list";

const mockStore = configureStore({});

describe(`Test 'OffersList '`, () => {
  it(`OffersList  should render correctly`, () => {
    const history = createMemoryHistory();
    const places = [{
      title: `Nice, cozy, warm big bed apartment`,
      price: 129,
      previewImage: ``,
      type: `room`,
      isFavorite: false,
      isPremium: false,
      rating: 3,
      id: 17,
      bedrooms: 1,
      description: ``,
      maxAdults: 1,
      location: {},
      images: [``],
      host: {id: 25, name: `Angelina`, isPro: false, avatarUrl: `img/avatar-angelina.jpg`},
      goods: [`Washer`],
      city: {name: `Amsterdam`, location: {}},
    }];

    const store = mockStore({
      PLACE_INFO: {placeInfo: places[0]},
      USER: {isLoggedIn: true}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <OffersList
              places={places}
              placesListName={CardsListName.NEAR_PLACES_LIST}
            />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
