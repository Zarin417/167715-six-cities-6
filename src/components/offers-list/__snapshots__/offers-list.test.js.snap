// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'OffersList ' OffersList  should render correctly 1`] = `
<div>
  <div
    class="near-places__list places__list"
  >
    <article
      class="near-places__card place-card"
    >
      <div
        class="near-places__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/17"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src=""
            width="260"
          />
        </a>
      </div>
      <div
        class=" place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              129
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button
       button"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 60%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/17"
          >
            Nice, cozy, warm big bed apartment
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          Room
        </p>
      </div>
    </article>
  </div>
</div>
`;
